src/database/schema/UserSchema.ts

```diff
@@ -305,11 +305,52 @@ class UserSchema implements BaseDbSchema {
         id: uuidv4(),
         channelId,
         userId: json?.user?.id,
-        username: json?.user?.name,
+        username: json?.user?.username,
         countryCode: '',
         createdAt: json?.user?.created_at,
         updatedAt: json?.updated_at,
         lastActiveAt: json?.user?.last_active,
+        profilePicture: json?.user?.image || DEFAULT_PROFILE_PIC_PATH,
+        bio: '',
+        isBanned: json?.banned,
+        anon_user_info_color_code: json?.anon_user_info_color_code,
+        anon_user_info_color_name: json?.anon_user_info_color_name,
+        anon_user_info_emoji_code: json?.anon_user_info_emoji_code,
+        anon_user_info_emoji_name: json?.anon_user_info_emoji_name
+      });
+    } catch (e) {
+      console.log('error on fromMemberWebsocketObject', e);
+      return new UserSchema({
+        id: uuidv4(),
+        channelId,
+        userId: null,
+        username: null,
+        countryCode: null,
+        createdAt: null,
+        updatedAt: null,
+        lastActiveAt: null,
+        profilePicture: null,
+        bio: null,
+        isBanned: null,
+        anon_user_info_color_code: null,
+        anon_user_info_color_name: null,
+        anon_user_info_emoji_code: null,
+        anon_user_info_emoji_name: null
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the catch block, you are returning a new UserSchema with all null values if there's an error. This could potentially lead to issues downstream where these values are expected to be non-null. It might be better to either throw the error after logging it or return a default user object with safe default values instead of nulls.



<!-- This is an auto-generated comment by OSS CodeRabbit -->